src/plasma/plasma_manager.cc

```diff
@@ -642,19 +647,16 @@ int read_object_chunk(ClientConnection *conn, PlasmaRequestBuffer *buf) {
   }
   r = read(conn->fd, buf->data + conn->cursor, s);
 
-  if (r == -1) {
-    LOG_ERROR("read error");
-  } else if (r == 0) {
-    LOG_DEBUG("end of file");
+  if (r <= 0) {
```

Comment:
Before it looks like `r == 0` meant we read an EOF. Is that an error?

Reply:
Yeah, it would be EOF for a normal file, but since this is over the network, and we stop reading after reading exactly the right number of bytes, we should never get 0 here, unless there was  anetwork error.